pkg/provisioners/channel_util.go

```diff
@@ -107,12 +123,16 @@ func createK8sService(ctx context.Context, client runtimeClient.Client, getSvc g
 	} else if err != nil {
 		return nil, err
 	}
-
 	// spec.clusterIP is immutable and is set on existing services. If we don't set this
 	// to the same value, we will encounter an error while updating.
-	svc.Spec.ClusterIP = current.Spec.ClusterIP
+	if svc.Spec.Type != corev1.ServiceTypeExternalName {
```

Comment:
Why not compare to `corev1.ServiceTypeClusterIP`?

Reply:
My understanding is that all other types except for ExternalName will have clusterIP immutable field. Isn't that so?